### PR TITLE
Sorbet and more

### PR DESCRIPTION
**Pull request by burke**
This will be a major version bump.

* Bumps minimum ruby version to 2.7 (and replaces 2.6 CI with 3.0)
* Removes the `Enumerable#min_by` backport, which exists since at least 2.5
* Adds sorbet with `typed: true` minimum, and signatures everywhere (currently all `T.untyped`)
* Brings in rubocop sorbet stuff
* Adds a sorbet runtime stub to avoid runtime deps
* Removes autocall, which I don't think I can feasibly make play nice with sorbet, and which delivered very little value.
* Also ran `rubocop -a`, which probably changed some incidental stuff.
* Makes the `dev.yml` commands a bunch more useful

From here we can start actually filling out the signatures with better than `T.untyped` but I think this is worth merging first.

I don't know how to make CI pass on the cursed platform.

### CODE REVIEW

**Comment by lugray** [comment]
How does a sorbetized gem interact with use in an app that doesn't otherwise use sorbet?

**Comment by burke** [comment]
In this case, I've stubbed out all the `T*` stuff by requiring `cli/kit/sorbet_runtime_stub` at the top of `cli/kit` if `!defined?(T)`, so all the sorbet stuff will noop at runtime.

Sorbet folks suggested rewriting for distribution with all the sorbet stuff stripped but this seems less magical and easier for now.

File: gen/lib/gen/commands.rb [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
       default: 'help',
       contextual_resolver: nil
     )
 
+    sig { params(const: T.untyped, cmd: T.untyped, path: T.untyped).returns(T.untyped) }
     def self.register(const, cmd, path)
       autoload(const, path)
+      # rubocop:disable Sorbet/ConstantsFromStrings
       Registry.add(->() { const_get(const) }, cmd)
+      # rubocop:enable Sorbet/ConstantsFromStrings
```
Comment (lugray) [inline gen/lib/gen/commands.rb:18]:
```suggestion
    sig { params(const: Symbol, cmd: String, path: String, lamda_const: T.proc.returns(Runtime::Command)).void }
    def self.register(const, cmd, path, lamda_const)
      autoload(const, path)
      Registry.add(lamda_const, cmd)
```

File: gen/lib/gen/commands.rb [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
     end
 
     register :Help, 'help', 'gen/commands/help'
```
Comment (lugray) [inline gen/lib/gen/commands.rb:21]:
```suggestion
    register :Help, 'help', 'gen/commands/help', -> { Help }
```

**Review by lugray — CHANGES_REQUESTED** [review]
Make register a bit more verbose, and Sorbet truly happy.

**Pull request merged**